app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,54 @@
+<section class='recipe'>
+  <h1>Recipe <%= params[:id] %></h1>
+  <div class="recipe-info">
+    <div>
+      <p>Preparation time: <%= @recipe.preparation_time %> H</p>
+      <p>Cooking time: <%= @recipe.cooking_time %> hours </p>
+      <% if @ispublic  %>
+        <p>Description:  </p> <br>
+        <p><%= @recipe.description %></p>
+      <%end%>
+    </div>
+    <div>
+    <% if can? :update, @recipe %>
+      <div class="d-flex flex-column">
+        <%= form_with model: @recipe, url: "/recipes/#{@recipe.id}", method: :put do |f| %>
+          <div class="form-check custom-control-right form-switch form-switch-sm mx-3">
+            <%= f.label :public, 'Public:', class: "h5 form-check-label", for: "flexSwitchCheckDefault" %>
+            <%= f.check_box :public, class: "form-check-input", id: "flexSwitchCheckDefault", onchange: "this.form.submit();" %>
+          </div>
+        <% end %>
+      </div>
+    <% end %>
+    </div>
+  </div>
+
+  <div class="btns-container">
+    <div class="m-4">
+      <%= button_to 'Generate shopping list', '/foods/new', method: :get %>
+    </div>

```

Comment:
- [ ] On submission of this form, the foods list does not get updated i.e. the new entry does not appear in the general shopping list. Give this a fix.